### PR TITLE
fix issue #1597: mark files being written as "active"

### PR DESCRIPTION
**Pull request by facontidavide**
**Suggested changes:**

1. Add the suffix ".active" to ` SequentialWriter::format_storage_uri`
2. When `storage_` is closed, rename the file to remove the ".active" suffix.
3. Save the metadata when splitting. In this way all the parts but the last one are still usable in case of crash.

To address #1597 point 3 is not necessary and **maybe** not even desirable, so let me know what you think.

**Note:** 

Since` rcpputils::fs` is not wrapping "rename", I had to use `std::filesystem` instead and upgrade to C++ 17.

### CODE REVIEW

File: rosbag2_cpp/src/rosbag2_cpp/writers/sequential_writer.cpp [hunk 1]
```diff
@@ -269,11 +287,22 @@ void SequentialWriter::switch_to_next_storage()
...
+  // Write the metadata before splitting. This will be replaced by an updated file later.
+  // In this way, if the process crashes, the metadata for the previous parts will still be available.
+  finalize_metadata();
+  metadata_io_->write_metadata(base_folder_, metadata_);
 
   storage_options_.uri = format_storage_uri(
     base_folder_,
     metadata_.relative_file_paths.size());
+  
+  // when the storage_ is closed, rename the file to remove ".inactive" suffix
```
Comment (facontidavide) [inline rosbag2_cpp/src/rosbag2_cpp/writers/sequential_writer.cpp]:
```suggestion
  // when the storage_ is closed, rename the file to remove ".active" suffix
```

**Comment by fujitatomoya** [comment]
@facontidavide thanks for the PR.

@MichaelOrlov are we supposed to create PR target to rolling 1st? and then backport?

**Comment by clalancette** [comment]
> @MichaelOrlov are we supposed to create PR target to rolling 1st? and then backport?

Yes please.  Also, we are going to run into other problems with the switch to `std::filesystem` on Humble (because of older GCC compilers).  But let's target this to `rolling` first, then deal with that in the backport.

**Comment by facontidavide** [comment]
I am temporarily closing this in favor of #1672

**Pull request closed without merging**